WebRoot/public/login.jsp

```diff
@@ -459,6 +459,13 @@ if (application.getInitParameter("offlineMode") != null) {
 					<c:otherwise>
 								<form method="post" name="loginForm" action="${formActionUrl}" accept-charset="UTF-8">
 								<input type="hidden" name="loginOp" value="login"/>
+
+                                <c:choose>
+                                    <c:when test="${totpAuthRequired || errorCode eq 'account.TWO_FACTOR_AUTH_FAILED'}">
```

Comment:
Why are we remembering the flag in case of TWO_FACTOR_AUTH_FAILED? 

Suggestion - Break the comment into multiple lines.